docs/examples/custom-json/README.md

```diff
@@ -0,0 +1,307 @@
+# Serving a custom model with JSON serialization
+
+The `mlserver` package comes with inference runtime implementations for `scikit-learn` and `xgboost` models.
+However, some times we may also need to roll out our own inference server, with custom logic to perform inference.
+To support this scenario, MLServer makes it really easy to create your own extensions, which can then be containerised and deployed in a production environment.
+
+## Overview
+
+In this example, we create a simple `Hello World JSON` model that parses and modifies a JSON data chunk. This is often useful as a means to
+quickly bootstrap existing models that utilize JSON based model inputs.
+
+## Serving
+
+The next step will be to serve our model using `mlserver`. 
+For that, we will first implement an extension which serve as the _runtime_ to perform inference using our custom `Hello World JSON` model.
+
+### Custom inference runtime
+
+This is a trivial model to demonstrate how to conceptially work with JSON inputs / outputs. In this example:
+
+- Parse the JSON input from the client
+- Create a JSON response echoing back the client request as well as a server generated message
+
+
+
+```python
+%%writefile models.py
+import json
+
+from typing import Dict, Any
+from mlserver import MLModel, types
+from mlserver.codecs import StringCodec
+
+from mlserver.cli.main import main
+
+
+class JsonHelloWorldModel(MLModel):
+    async def load(self) -> bool:
+        # Perform additional custom initialization here.
+        print("Initialize model")
+
+        # Set readiness flag for model
+        return await super().load()
+    
+    async def predict(self, payload: types.InferenceRequest) -> types.InferenceResponse:
+        request = self._extract_json(payload)
+        response = { 
+            "request": request,
+            "server_response": f"Got your request. Hello from the server."
+        }
+        response_bytes = json.dumps(response).encode("UTF-8")
+
+        return types.InferenceResponse(
+            id=payload.id,
+            model_name=self.name,
+            model_version=self.version,
+            outputs=[
+                types.ResponseOutput(
+                    name="echo_response",
+                    shape=[len(response_bytes)],
+                    datatype="BYTES",
+                    data=[response_bytes],
+                    parameters=types.Parameters(content_type="str")
+                )
+            ]
+        )
+
+    def _extract_json(self, payload: types.InferenceRequest) -> Dict[str, Any]:
+        inputs = {}
+        for inp in payload.inputs:
+            inputs[inp.name] = json.loads("".join(self.decode(inp)))
```

Comment:
Would it make sense to pass the `default_codec` kwarg on the `self.decode()` call to ensure its not dependant on the `content_type` being set?